### DISCUSSION TITLE
Is it possible to load images directly onto another image?

### DISCUSSION THREAD

**Discussion by yt6983138** [SixLabors/ImageSharp#2832] [Q&A]
Hello, I'm making a html rendering tool recently, and having need to combine few images together, currently I do it like this:
```c#
using Image<Rgba32> bigImage = new(width, height);

// note, i had to use block based because chromium is not able to handle large images (repeating image when dim reaches 16384,
// and random unrendered block when size > 2k*8k)
for (int x = 0; x < (width / BlockSize) + 1; x++) 
{
	for (int y = 0; y < (height / BlockSize) + 1; y++)
	{
		int vpX = x * BlockSize;
		int vpY = y * BlockSize;
		byte[] raw = /* get the raw image somewhere... */;

		using Image rawImage = Image.Load(raw);
		bigImage.Mutate(x => x.DrawImage(rawImage, new Point(vpX, vpY), 1));
	}
}
// save the bigImage...
```
But as I measured, this is extremely memory inefficient, as it has 2 (3) copy of the image block, the raw image, the decoded image, and the big image.

I wonder is it possible to directly load the image onto the big image? Like once I get the image block then I load it onto the bigger image directly, no need for creating copies of the image block

**Comment by JimBobSquarePants** [comment]
I see from your code, the big image, the decoded image block and the encoded bytes for the block in an unknown format. I'mnot sure how you intend to load the block image without decodeing those bytes.  

Can your `byte[]` not be a stream source? That way you can manage the raw encoded bytes without each allocation to a `byte[]`

**Reply by yt6983138** [reply]
Thanks for the reply, sorry I didn't clarify; I mean can I just decode the image block onto the bigger image without creating copies of the image block? 
Also, each image block is a complete image (with metadata etc, like a complete png/jpg/webp), so is there something I can do like, decoding the pixels one by one and writing to the memory of the bigger image? 
I also have control of the image format, it must be one of png, jpeg, and webp (it is specified by the parameter)
